### PR TITLE
Handle answering a question

### PR DESCRIPTION
**Pull request by aarondufall**

### CODE REVIEW

File: lib/question_component/question.rb [hunk 1]
```diff
@@ -7,9 +7,14 @@ class Question
     attribute :details, String
     attribute :tags, Array
     attribute :asked_time, Time
+    attribute :answers, Array, default: []
 
     def asked?
       !asked_time.nil?
     end
+
+    def answer(new_answer)
```
Comment (aarondufall) [inline lib/question_component/question.rb]: Is this the best interface for the kind of behaviour

File: lib/question_component/question/answer.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module QuestionComponent
+  class Question
+    class Answer
```
Comment (aarondufall) [inline lib/question_component/question/answer.rb:3]: Should this be namespaced under `Question`

File: lib/question_component/projection.rb [hunk 3]
```diff
@@ -17,5 +17,20 @@ class Projection
...
+        { :answer_id => :id },
+        :body
+      ])
+
+      answered_time = Clock.parse(answered.time)
+
+      answer.answered_time = answered_time
+
+      question.answer(answer)
+    end
```
Comment (aarondufall) [inline lib/question_component/projection.rb]: Is this the correct way to apply a child object?

File: lib/question_component/question.rb (hunk 1 above)
Comment (ntl) [reply lib/question_component/question.rb]: I think you'd want each of the primitive attributes of a `Question::Answer` data structure to be arguments of this method. That way this class can encapsulate the constant look up of the subordinate "child" class.

File: lib/question_component/question/answer.rb (hunk 2 above)
Comment (ntl) [reply lib/question_component/question/answer.rb:3]: If it's a data structure that is always found as a child object of a question, it should be here.

**Pull request merged**